lib/awful/util.lua

```diff
@@ -253,16 +253,7 @@ end
 -- @tparam string path
 -- @treturn bool True if path exists and is a directory.
 function util.is_dir(path)
-    local file = io.open(path)
-    if file then
-        if not file:read(0) -- Not a regular file (might include empty ones).
-            and file:seek("end") ~= 0 then  -- And not a file with size 0.
-            io.close(file)
-            return true
-        end
-        io.close(file)
-    end
-    return false
+    return Gio.File.new_for_path(path):query_file_type({}) == "DIRECTORY"
```

Comment:
No, that's ok. I don't know if lgi even let's you access this function. However, a `GFile` only contains a path name. It doesn't keep the file open or anything like that. Thus, this problem doesn't apply here.
(It's similar to "always remember to call `io.close` on file handles in Lua)